test/dynamo/test_dynamic_shapes.py

```diff
@@ -25,21 +25,83 @@
 import unittest
 
 
-def make_dynamic_cls(cls):
+def make_dynamic_cls(cls, assume_static_by_default):
+    assume_static_by_default_suffix = (
+        "_static_default" if assume_static_by_default else ""
+    )
     return make_test_cls_with_patches(
-        cls, "DynamicShapes", "_dynamic_shapes", (config, "dynamic_shapes", True)
+        cls,
+        "DynamicShapes",
+        f"_dynamic_shapes{assume_static_by_default_suffix}",
+        (config, "dynamic_shapes", True),
+        (config, "assume_static_by_default", assume_static_by_default),
     )
 
 
-DynamicShapesFunctionTests = make_dynamic_cls(test_functions.FunctionTests)
-DynamicShapesMiscTests = make_dynamic_cls(test_misc.MiscTests)
-DynamicShapesReproTests = make_dynamic_cls(test_repros.ReproTests)
-DynamicShapesNNModuleTests = make_dynamic_cls(test_modules.NNModuleTests)
-DynamicShapesUnspecTests = make_dynamic_cls(test_unspec.UnspecTests)
-DynamicShapesExportTests = make_dynamic_cls(test_export.ExportTests)
-DynamicShapesSubGraphTests = make_dynamic_cls(test_subgraphs.SubGraphTests)
+# Note - this is a little redundant, but some of these tests have specific
+# exclusions.
+# Note #2 - Putting this in a for loop over [True, False] for
+# assume_static_by_default clobbers the name, which makes the tests not run.
```

Comment:
You could do setattr(sys.module[__name__], blah) to programatically set the name

Reply:
Is it worth it, at that point? no strong opinion, this felt the most readable atm.

Reply:
My preference is still for looping, because whenever a new test file is added, in the current configuration you have to do it twice